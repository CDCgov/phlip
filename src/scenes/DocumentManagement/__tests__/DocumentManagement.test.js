import React from 'react'
import { shallow } from 'enzyme'
import { DocumentManagement } from '../index'

const props = {
  page: 0,
  rowsPerPage: '10',
  documents: [],
  allSelected: false,
  docCount: 0,
  actions: {
    getDocumentsRequest: jest.fn(),
    handleSelectAll: jest.fn(),
    handleSearchFieldChange: jest.fn(),
    handlePageChange: jest.fn(),
    handleRowsChange: jest.fn(),
    handleSelectOneFile: jest.fn()
  },
  apiErrorOpen: false,
  apiErrorInfo: {
        title: '',
        text: ''
    },
}

describe('Document Management scene', () => {
  test('should render correctly', () => {
    expect(shallow(<DocumentManagement {...props} />)).toMatchSnapshot()
  })
})