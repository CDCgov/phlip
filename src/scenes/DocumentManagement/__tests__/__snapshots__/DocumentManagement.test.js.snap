// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Document Management scene should render correctly 1`] = `
<FlexGrid
  align="stretch"
  circular={false}
  container={true}
  flex={true}
  justify="stretch"
  padding="20px 30px"
  raised={false}
  style={Object {}}
  type="column"
>
  <withRouter(WithTheme(PageHeader))
    entryScene={true}
    icon="description"
    otherButton={
      Object {
        "isLink": true,
        "path": "/docs/upload",
        "props": Object {
          "aria-label": "Upload New Documents",
        },
        "show": true,
        "state": Object {
          "modal": true,
        },
        "text": "+ Upload New",
      }
    }
    pageTitle="Document Management"
    projectName=""
    protocolButton={false}
  />
  <FlexGrid
    align="stretch"
    circular={false}
    container={true}
    flex={true}
    justify="stretch"
    padding={0}
    raised={true}
    style={Object {}}
    type="column"
  >
    <Search
      onClearSuggestions={[Function]}
      onGetSuggestions={[Function]}
      onSearchDocs={[Function]}
      onSearchValueChange={[Function]}
      onSuggestionSelected={[Function]}
    />
    <DocList
      allSelected={false}
      docCount={0}
      documents={Array []}
      onChangePage={[MockFunction]}
      onChangeRows={[MockFunction]}
      onSelectAllFiles={[MockFunction]}
      onSelectOneFile={[MockFunction]}
      page={0}
      rowsPerPage="10"
    />
  </FlexGrid>
  <Route
    component={[Function]}
    path="/docs/upload"
  />
</FlexGrid>
`;
